tuf/tuf.go

```diff
@@ -202,6 +223,47 @@ func (tr *Repo) UpdateDelegations(role *data.Role, keys []data.PublicKey, before
 
 	tr.keysDB.AddRole(role)
 
+	utils.RemoveUnusedKeys(p)
+
+	return nil
+}
+
+// DeleteDelegation removes a delegated targets role from its parent
+// targets object. It also deletes the delegation from the snapshot.
+// DeleteDelegation will only make use of the role Name field.
+func (tr *Repo) DeleteDelegation(role data.Role) error {
```

Comment:
Should this also remove any child roles?  Or error maybe if a child role is then orphaned?